File: Part3/T3L90extra_ok.c
long int A,B;
unsigned char C;


void Print_one();
void Print_two();
void Print_three();
void Print_dunno();
void Print_big();
void Print_just();
void Print_isnt();
void Print_hello();
void Print_oops();
void Print_712();
void Print_is_the();
void Print_grooviest();
void Print_newline();


int main()
{
    A = 3;

    if (A>=5 && A<=10)
        Print_oops();
    else if (A>=1 && A<=3) {
        Print_one();
        Print_two();
        Print_three();
    }
    else
        Print_dunno();

    A = 4;

    if (A>=5 && A<=10)
        Print_big();
    else if (A>=1 && A<=3) {
        Print_one();
        Print_two();
        Print_three();
    }
    else
        Print_dunno();
           
    A = 7;

    if (A>=5 && A<=10)
        Print_big();
    else if (A>=1 && A<=3) {
        Print_one();
        Print_two();
        Print_three();
    }
    else
        Print_dunno();
           
    Print_newline();
    return 0;
}

File: test/T3L90extra_ok.pas
(* Test file for level 90 of CSCE 531 Pascal compiler Project III *)

(* There are no errors in this file *)

program T3L90extra_ok;

var
   a,b : Integer;
   c   : Char;


procedure print_one; External;
procedure print_two; External;
procedure print_three; External;
procedure print_dunno; External;
procedure print_big; External;
procedure print_just; External;
procedure print_isnt; External;
procedure print_hello; External;
procedure print_oops; External;
procedure print_712; External;
procedure print_is_the; External;
procedure print_grooviest; External;
procedure print_newline; External;


begin

   a := 3;

   case a of
     5..10: print_oops;
     1..3:
        begin   
	   print_one;
	   print_two;
	   print_three
	end
   else
      print_dunno
   end; { case }

   a := 4;

   case a of
     5..10: print_big;
     1..3:
        begin   
	   print_one;
	   print_two;
	   print_three
	end
   else
      print_dunno
   end; { case }
   
   a := 7;

   case a of
     5..10: print_big;
     1..3:
        begin   
	   print_one;
	   print_two;
	   print_three
	end
   else
      print_dunno
   end; { case }
   
   print_newline

end.
